src/api/organisationUnits.js

```diff
@@ -1,64 +1,115 @@
 import { onError } from './index'
 
-export const apiFetchOrganisationUnitRoot = d2 => {
-    const endPoint = '/organisationUnits'
-    const fields = ['id', 'displayName', 'name']
-    const url = `${endPoint}?paging=false&userDataViewFallback=true&fields=${fields.join(
-        ','
-    )}`
-
-    return d2.Api.getApi()
-        .get(url)
-        .then(({ organisationUnits }) => organisationUnits[0])
-        .catch(onError)
+const orgUnitLevelsQuery = {
+    resource: 'organisationUnitLevels',
+    params: ({ displayNameProp = 'displayName' }) => ({
+        fields: `id,level,${displayNameProp}~rename(displayName),name`,
+        paging: false,
+    }),
 }
 
-/**
- * Fetch organisation units
- * @returns {Promise<T | never>}
- */
-export const apiFetchOrganisationUnits = (d2, displayNameProperty) => {
-    const fields = [
-        'id',
-        'path',
-        `${displayNameProperty}~rename(displayName)`,
-        'children::isNotEmpty',
-    ]
+const orgUnitGroupsQuery = {
+    resource: 'organisationUnitGroups',
+    params: ({ displayNameProp = 'displayName' }) => ({
+        fields: `id,${displayNameProp}~rename(displayName),name`,
+        paging: false,
+    }),
+}
 
-    return d2.models.organisationUnits.list({
+const orgUnitRootsQuery = {
+    resource: 'organisationUnits',
+    params: {
+        fields: 'id,displayName,name',
+        userDataViewFallback: true,
         paging: false,
+    },
+}
+
+const orgUnitsQuery = {
+    resource: 'organisationUnits',
+    params: ({ displayNameProp }) => ({
+        fields: `id,path,${displayNameProp}~rename(displayName),children::isNotEmpty`,
         level: 1,
```

Comment:
This `level: 1,` was already there so it must be correct, but I don't understand how having a hardcoded `level` is present if this is a generic way to get a list of org-units.

Reply:
I think this fetches the "highest level" for the user (which is dynamic, could be one level for you and another for me, depending on our root org units). Not entirely sure, but I believe @janhenrikoverland knows more about how the backend treats this.